lib/snowpack/type.ex

```diff
@@ -137,12 +138,33 @@ defmodule Snowpack.Type do
   @doc """
   Transforms `:odbc` return values to Elixir representations.
   """
-  @spec decode(:odbc.value(), opts :: Keyword.t()) :: return_value()
+  @spec decode(value(), opts :: Keyword.t()) :: return_value()
   def decode(:null, _), do: nil
 
-  def decode(value, _opts), do: value
+  def decode(value, _) when is_float(value), do: value
+  def decode(value, _) when is_boolean(value), do: value
+  def decode(value, _) when is_integer(value), do: value
 
-  defp to_encoded_string(data), do: to_string(data)
+  def decode({{_year, _month, _day}, {_hour, _minute, _sec}} = value, _) do
+    case NaiveDateTime.from_erl(value) do
+      {:ok, datetime} -> datetime
+      {:error, _error} -> value
+    end
+  end
+
+  def decode(value, _) when is_binary(value) do
+    parse_funcs = [&Integer.parse/1, &DateTimeParser.parse/1]
+
+    Enum.find_value(parse_funcs, value, fn func ->
+      case Kernel.apply(func, [value]) do
+        {integer, ""} -> integer
+        {:ok, result} -> result
+        _ -> false
+      end
+    end)
+  end
+
+  def decode(value, _opts), do: value
```

Comment:
If the type parsing fails for some reason these `decode/2` are called by `DBConnection` and can make a best attempt for types coming from Erlang `:odbc`.